biogrid/apps/biogrid-mvp/src/app/pages/input-page/input-page.css

```diff
@@ -31,38 +36,237 @@ br {
   width: 56.25rem;
 }
 
+#intro {
+  background: #adadad;
+  border: 5px solid black;
+  border-radius: 5px;
+  color: rgb(0, 0, 0);
+  padding: 10px;
+}
+
 label {
   display: block;
   text-align: left;
 }
 
 .largeBatteryCells {
   display: inline-block;
+  margin-bottom: 1.25rem;
+  margin-left: 1.25rem;
+  margin-right: 1.25rem;
+  margin-top: 1.25rem;
 }
 
 .smallBatteryCells {
   display: inline-block;
-  margin-right: 1.25rem;
   margin-bottom: 1.25rem;
+  margin-left: 1.25rem;
+  margin-right: 1.25rem;
+  margin-top: 1.25rem;
 }
 
 .solarPanels {
   display: inline-block;
+  margin-bottom: 1.25rem;
+  margin-left: 1.25rem;
+  margin-right: 1.25rem;
+  margin-top: 1.25rem;
 }
 
 .startDatePicker {
   display: inline-block;
-  margin-top: 1.25rem;
-  margin-right: 1.25rem;
   margin-bottom: 1.25rem;
+  margin-left: 1.25rem;
+  margin-right: 1.25rem;
+  margin-top: 1.25rem;
 }
 
 .submitButton {
   text-align: center;
 }
 
-.townSize {
+.tooltipBuild {
+  position: relative;
+  display: inline-block;
+}
+
+.tooltipBuild .tooltiptextBuild {
+  visibility: hidden;
+  width: 450px;
+  background-color: rgb(22, 22, 22);
+  color: #fff;
+  text-align: center;
+  border-radius: 6px;
+  padding: 5px 0;
+  position: absolute;
+  z-index: 1;
+}
+
+.tooltipBuild:hover .tooltiptextBuild {
+  visibility: visible;
+}
+
+
+.tooltipEndDate {
+  position: relative;
+  display: inline-block;
+}
+
+.tooltipEndDate .tooltiptextEndDate {
+  visibility: hidden;
+  width: 450px;
+  background-color: rgb(22, 22, 22);
+  color: #fff;
+  text-align: center;
+  border-radius: 6px;
+  padding: 5px 0;
+  position: absolute;
+  z-index: 1;
+}
+
+.tooltipEndDate:hover .tooltiptextEndDate {
+  visibility: visible;
+}
+
+.tooltipHeight {
+  position: relative;
+  display: inline-block;
+}
+
+.tooltipHeight .tooltiptextHeight {
+  visibility: hidden;
+  width: 450px;
+  background-color: rgb(22, 22, 22);
+  color: #fff;
+  text-align: center;
+  border-radius: 6px;
+  padding: 5px 0;
+  position: absolute;
+  z-index: 1;
+}
```

Comment:
same with all the tooltip stuff - you can re-use one class for each tooltip and only have to write this definition once

Reply:
+1. you can condense about some classes to one. For instance where you change the visibility to visible, just make it one class such
`.addVisibility {
   visibility: visible;
}`
`.hideVisibility {
   visibility: hidden;
}`
Then you can reuse these two classes whenever you want to change the visibility.
This is just an example and might not necessarily fit into this specific code